core/controllers/profile.py

```diff
@@ -309,6 +425,44 @@ def get(self):
 class SignupHandler(base.BaseHandler):
     """Provides data for the editor prerequisites page."""
 
+    URL_PATH_ARGS_SCHEMAS = {}
+    HANDLER_ARGS_SCHEMAS = {
+        'GET': {},
+        'POST': {
+            'username': {
+                'schema': {
+                    'type': 'basestring',
+                    'validators': [{
+                        'id': 'is_valid_username_string'
+                    }]
+                },
+                'default_value': None
+            },
+            'agreed_to_terms': {
+                'schema': {
+                    'type': 'bool'
+                },
+                'default_value': False
+            },
+            'default_dashboard': {
+                'schema': {
+                    'type': 'basestring',
+                    'choices': [
+                        constants.DASHBOARD_TYPE_LEARNER,
+                        constants.DASHBOARD_TYPE_CREATOR
+                    ]
+                },
+                'default_value': None
+            },
+            'can_receive_email_updates': {
+                'schema': {
+                    'type': 'bool'
+                },
+                'default_value': None
```

Comment:
Why `None`?

Reply:
I found that it is optional in the request, that's why I added `None` as the `default` value. Please tell me if I did it in the wrong way.

Reply:
Again, please link the code that does this.

Reply:
[Here](https://github.com/oppia/oppia/blob/develop/core/templates/pages/signup-page/services/signup-page-backend-api.service.ts#:~:text=export%20interface%20UpdateUsernameRequestParams,%7D) I found `username` is not optional and as you have suggested, I have removed the `default_value` and updated a few backend tests.

Reply:
But what about this and the other one you kept as `None`? They also always seem to be passed.

Reply:
[Here](https://github.com/oppia/oppia/blob/2c4ce8ed56e0e605c3f4637337651d7a2cf31fd0/core/controllers/profile.py#L341) it seems like `can_receive_email_updates` can have none. Also, I tried removing none here but that was giving a 500 error. So, I think this arg is optional and for that, I added `none` in `default_value`.